lib/resources.js

```diff
@@ -605,6 +603,86 @@ const buildElasticQueryForKeywords = function (params) {
   return query
 }
 
+const filterClausesForDateParams = function (filters) {
```

Comment:
A couple of comments below. This is fine but I think it could be DRYed up.

```
let must = [ { exists: { field: 'dateStartYear' } }]
if (hasDateBefore) {
    must.push({
        { bool: { should: [ { range: ...
}
if (hasDateAfter) {
    must.push( yadda yadda yadda
}
return [{ clause: { bool: must } }]
```

Reply:
I have pushed a change to DRY it up a bit. I think DRY sometimes saves on application size but reduces readability (without appreciably improving maintainability), which was my concern here, but let me know if you think the update improves things.

Reply:
Hmmm... I think the issue is less about saving space and more about having each logical requirement enforced by a single bit of code and vice versa. In this case, there are two different places managing the `dateBefore` part of the query, but it seems like maybe you think these are actually sufficiently different cases that it should be handled separately? That seems like a judgment call and I am happy enough to follow you on it. 
Not to be a pain but I don't think this is any better than the way it was previously. If you think the two cases should really be separate I would probably prefer going back to what it was before.

Reply:
I have pushed another attempt to improve things DRY-wise. Previously I was thinking of the three possible combinations of params as necessitating three substantially different filter queries. They still kind of do, but I think this version does a better job isolating those differences (and ignoring one of the less important differences). Let me know what you think

Reply:
This looks, great, thanks!